### PR TITLE
Integrate pre-commit

### PR DESCRIPTION
**Pull request by domenkozar**
Need to reformat everything, so let's wait until PR queue is empty

### CODE REVIEW

File: hercules-ci-agent/pkg.nix [hunk 1]
```diff
@@ -1,15 +1,63 @@
...
+, text
+, time
+, tomland
+, transformers
+, transformers-base
+, unix
+, unliftio
+, unliftio-core
+, unordered-containers
+, uuid
```
Comment (roberth) [inline hercules-ci-agent/pkg.nix]: Not sure if desirable. Perhaps `generate-nix` should call the formatter.

File: hercules-ci-agent/hercules-ci-agent-worker/CNix.hs [hunk 2]
```diff
@@ -95,16 +107,21 @@ withEvalState
   => Ptr (Ref NixStore)
   -> (Ptr EvalState -> ConduitT i o m r)
   -> ConduitT i o m r
-withEvalState store = bracketP (liftIO $ [C.throwBlock| EvalState* {
+withEvalState store =
+  bracketP
+    ( liftIO
+        $ [C.throwBlock| EvalState* {
       Strings searchPaths;
```
Comment (roberth) [inline hercules-ci-agent/hercules-ci-agent-worker/CNix.hs]: This indentation is wrong, but can presumably be fixed manually.
Other places need it too. Look for `|]`.

File: tests/agent-test/testdata/ambiguous-nix-file/ci.nix [hunk 3]
```diff
@@ -1 +1 @@
-null
```
Comment (roberth) [inline tests/agent-test/testdata/ambiguous-nix-file/ci.nix]: Formatting test data is asking for trouble. Can we exclude `tests/agent-test/testdata`?

File: hercules-ci-agent/pkg.nix (hunk 1 above)
Comment (domenkozar) [reply hercules-ci-agent/pkg.nix]: This will be gone with haskell.nix, so let's ignore formatting for now to avoid conflicts

Comment (domenkozar) [reply hercules-ci-agent/pkg.nix]: I've let this be, it's going to be easier to just delete these files once we get there.

For now the workflow is the same as now, you write changes and run pre-commit to format.

**Pull request merged**